cpp/tensorrt_llm/plugins/mixtureOfExperts/mixtureOfExpertsPlugin.cpp

```diff
@@ -396,16 +402,23 @@ bool MixtureOfExpertsPlugin::supportsFormatCombination(
 
     if (pos == getExpertWeights1Index() || pos == getExpertWeights2Index())
     {
-        if (mGroupwiseQuantAlgo == 0)
+        // FP4 and INT4 weights must be packed as INT8 to the plugin because
+        // TensorRT does not support these types.
+        if (mWeightType == nvinfer1::DataType::kINT4 || mWeightType == nvinfer1::DataType::kFP4)
         {
-            auto normalized_weight_type
-                = mWeightType == nvinfer1::DataType::kINT4 ? nvinfer1::DataType::kINT8 : mWeightType;
-            return inOut[pos].type == normalized_weight_type;
+            return inOut[pos].type == nvinfer1::DataType::kINT8;
```

Comment:
how does this work? If the weights are int4 or fp4 quantized, we have to provide the weights as int8 tensors? So the weight would be int8 and the `weightType` parameter would have to be fp4?

Reply:
yes. that's right.